lib/internal/webstreams/adapters.js

```diff
@@ -428,13 +428,21 @@ function newReadableStreamFromStreamReadable(streamReadable, options = kEmptyObj
 
   let controller;
 
-  function onData(chunk) {
-    // Copy the Buffer to detach it from the pool.
-    if (Buffer.isBuffer(chunk) && !objectMode)
-      chunk = new Uint8Array(chunk);
-    controller.enqueue(chunk);
-    if (controller.desiredSize <= 0)
-      streamReadable.pause();
+  function fillData() {
+    if (controller.desiredSize <= 0) {
+      return;
+    }
+
+    let chunk;
+    while ((chunk = streamReadable.read(controller.desiredSize)) !== null) {
```

Comment:
Need to make sure `controller.desiredSize`  must be less than 1GB as the docs specify

> [...] The size argument must be less than or equal to 1 GiB. [...]
>
> from [`readable.read([size])` - Node Docs](https://nodejs.org/api/stream.html#readablereadsize)

Reply:
Should I don't specify `size` argument here, in case the `desiredSize` is really big?